param/parameterized.py

```diff
@@ -54,6 +54,24 @@ def get_logger():
 warning_count = 0
 
 
+def create_label_formatter(capitalize=True, replace_underscores=True, overrides={}):
+    """
+    Utility to create a simple, custom label formatter.
+    """
+    def label_formatter(pname):
+        if pname in overrides:
+            return overrides[pname]
+        if replace_underscores:
+            pname = pname.replace('_',' ')
+        if capitalize:
+            pname = pname[:1].upper() + pname[1:]
+        return pname
+    return label_formatter
+
+
+label_formatter = create_label_formatter()
```

Comment:
Just to check my understanding, it looks like this is creating a closure that stores the state of `replace_underscores` and `overrides`, and that the reason it's done this way as opposed to the usual approach of a ParameterizedFunction is to avoid circularity, in that this label_formatter is used in the definition of Parameter?  If so that's ok.  Not sure if it would be cleaner to define it as a ParameterizedFunction but only set the value on Parameter once Parameterized has been defined; the advantage would be that then one could do `LabelFormatter.replace_underscores=False`, but the disadvantage would be the difficult to track circularity.

Reply:
> Just to check my understanding, it looks like this is creating a closure that stores the state of replace_underscores and overrides, and that the reason it's done this way as opposed to the usual approach of a ParameterizedFunction is to avoid circularity, in that this label_formatter is used in the definition of Parameter? 

That is exactly it. The original *was* a parameterized function that I tried to preserve but the circularity was a problem (also, it needs the parameters themselves which are in the other file, causing circular imports).

Reply:
I switched back to `ParameterizedFunction` in efa8343 but only because I am using `param.Parameter` instead of the more appropriate parameter types...

Reply:
Looks good.